spectrum_utils/plot.py

```diff
@@ -227,6 +248,129 @@ def spectrum(
+        the unit that was used for spectrum annotation is used. The default is
+        None.
+    plot_unknown : bool, optional
+        Flag indicating whether or not to plot mass errors for unknown peaks.
+    color_ions : bool, optional
+        Flag indicating whether or not to color dots for annotated fragment
+        ions. The default is True.
+    grid : Union[bool, str], optional
+        Draw grid lines or not. Either a boolean to enable/disable both major
+        and minor grid lines or 'major'/'minor' to enable major or minor grid
+        lines respectively.
+    ax : Optional[plt.Axes], optional
+        Axes instance on which to plot the mass errors. If None the current
+        Axes instance is used.
+
+    Returns
+    -------
+    plt.Axes
+        The matplotlib Axes instance on which the mass errors are plotted.
+
+    Notes
+    -----
+    The mass error bubble plot was first introduced in [1]_.
+
+    References
+    ----------
+    .. [1] Barsnes,H., Eidhammer,I. and Martens,L. (2010)
+       FragmentationAnalyzer: An open-source tool to analyze MS/MS
+       fragmentation data. PROTEOMICS, 10, 1087–1090.
+       doi:10.1002/pmic.200900681
+
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    _format_ax(ax, grid)
+
+    if len(spec.mz) == 0:
+        ax.set_ylabel("Mass error")
+        ax.set_ylim(-1, 1)
+        return ax
+
+    annotations = (
+        spec.annotation
+        if spec.annotation is not None
+        else itertools.repeat(None, len(spec.mz))
+    )
+
+    known_ions = []
+    dot_colors = []
+    mz_deltas = []
+    mz_delta_units = []
+    for ann in annotations:
+        # Use the first annotation in case there are multiple options.
+        ion_type = ann[0].ion_type[0] if ann is not None else None
+        is_known_ion = ion_type is not None and ion_type != "?"
+        known_ions.append(is_known_ion)
+        dot_colors.append(colors.get(ion_type if color_ions else None))
+        mz_deltas.append(ann[0].mz_delta[0] if is_known_ion else 0.0)
+        mz_delta_units.append(ann[0].mz_delta[1] if is_known_ion else None)
+
+    dot_colors = np.array(dot_colors)
+    mz_deltas = np.array(mz_deltas)
+    intensity_scaled = 500 * (spec.intensity / np.max(spec.intensity))
+    mask = (
+        np.ones_like(spec.mz, dtype=bool)
+        if plot_unknown
+        else np.array(known_ions)
+    )
+
+    for known_unit in ["ppm", "Da"]:
+        # Use `not any` instead of `all` to fail fast
+        if not any(u and u != known_unit for u in mz_delta_units):
+            annotation_unit = known_unit
+            break
+    else:
+        raise ValueError("Inconsistent or unknown mass units in annotations.")
+    if unit == "Da" and annotation_unit == "ppm":
+        mz_deltas = ppm_to_da(mz_deltas, spec.mz)
+    elif unit == "ppm" and annotation_unit == "Da":
+        mz_deltas = da_to_ppm(mz_deltas, spec.mz)
+
+    y_lim = 1.2 * np.max(np.abs(mz_deltas))
+    if y_lim != 0:
```

Comment:
**nitpick:** This is a float, so I think it's better to use `>` instead of `!=` to avoid weird floating point behavior.